.github/workflows/_classic-buildpack-prepare-release.yml

```diff
@@ -0,0 +1,117 @@
+name: Prepare release
+
+on:
+  workflow_call:
+    inputs:
+      app_id:
+        description: Application ID of GitHub application (e.g.; the Linguist App)
+        type: string
+        required: true
+      app_email:
+        description: The email address of the GitHub application bot user (e.g.; the Linguist App)
+        type: string
+        required: false
+        default: ${{ vars.LINGUIST_GH_APP_EMAIL }}
+      app_username:
+        description: The username of the GitHub application bot user (e.g.; the Linguist App)
+        type: string
+        required: false
+        default: ${{ vars.LINGUIST_GH_APP_USERNAME }}
+      buildpack_name:
+        description: The name of the buildpack we're preparing
+        type: string
+        required: true
+      ip_allowlisted_runner:
+        description: The GitHub Actions runner to use to run jobs that require IP allow-list privileges
+        type: string
+        required: false
+        default: pub-hk-ubuntu-22.04-small
+      unreleased_header_text:
+        description: The header used to indicate unreleased changes in the CHANGELOG.md
+        type: string
+        required: false
+        default: Unreleased
+    secrets:
+      app_private_key:
+        description: Private key of GitHub application (Linguist)
+        required: true
+
+defaults:
+  run:
+    # Setting an explicit bash shell ensures GitHub Actions enables pipefail mode too,
+    # ratherthan only error on exit (improving failure UX when pipes are used). See:
+    # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell
+    shell: bash
+
+jobs:
+  prepare-release:
+    name: Prepare Release
+    runs-on: ${{ inputs.ip_allowlisted_runner }}
+    steps:
+      - name: Get token for GH application (Linguist)
+        uses: heroku/use-app-token-action@main
+        id: generate-token
+        with:
+          app_id: ${{ inputs.app_id }}
+          private_key: ${{ secrets.app_private_key }}
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          # We always want the version bump/changelog and resultant PR to target main, not the branch of the workflow_dispatch.
+          ref: main
+          # Using the GH application token here will configure the local git config for this repo with credentials
+          # that can be used to make signed commits that are attributed to the GH application user
+          token: ${{ steps.generate-token.outputs.app_token }}
+
+      - name: Determine existing published version
```

Comment:
I have an alternative approach in #184, which uses git tags to calculate the existing version, and so works with Ruby and PHP too.

I've also added support for running a custom command before committing, so the Ruby buildpack can make it's additional edits. (Such as updating the buildpack version in [`language_pack/version.rb`](https://github.com/heroku/heroku-buildpack-ruby/blob/main/lib/language_pack/version.rb) plus moving the `changelogs/unreleased/` directory to `changelogs/vNNN/` etc)